Language/Haskell/GhcMod/Monad.hs

```diff
@@ -52,17 +52,17 @@ import Exception
 import System.Directory
 import Prelude
 
-withGhcModEnv :: (IOish m, GmOut m) => FilePath -> Options -> (GhcModEnv -> m a) -> m a
+withGhcModEnv :: (IOish m, GmOut m) => FilePath -> Options -> ((GhcModEnv, GhcModLog)  -> m a) -> m a
 withGhcModEnv = withGhcModEnv' withCradle
  where
    withCradle dir =
-       gbracket (findCradle' dir) (liftIO . cleanupCradle)
-
-withGhcModEnv' :: (IOish m, GmOut m) => (FilePath -> (Cradle -> m a) -> m a) -> FilePath -> Options -> (GhcModEnv -> m a) -> m a
+       gbracket (runJournalT $ findCradle' dir) (liftIO . cleanupCradle . fst)
```

Comment:
Well maybe you want to enable detailed logging for just one command in ghc-modi/legacy-interavtive or something ;) There's just no good reason not to allow it to change since it fits nicely into the Writer/GmLog thing.